Realm/RLMMigration.mm

```diff
@@ -157,20 +132,16 @@ - (BOOL)deleteDataForClassName:(NSString *)name {
         return false;
     }
 
-    size_t table = _realm.group->find_table(RLMStringDataWithNSString(RLMTableNameForClass(name)));
-    if (table == realm::not_found) {
+    TableRef table = ObjectStore::table_for_object_type(_realm.group, name.UTF8String);
+    if (!table) {
         return false;
     }
 
     if ([_realm.schema schemaForClassName:name]) {
-        _realm.group->get_table(table)->clear();
+        table->clear();
     }
     else {
-        _realm.group->remove_table(table);
-
-        if (RLMRealmPrimaryKeyForObjectClass(_realm, name)) {
```

Comment:
Will need to add this back in I guess. We can remove entries for any tables for deleted object types whether or not this method is called.